base/array.jl

```diff
@@ -1009,21 +1013,33 @@ end
 
 # efficiently grow an array
 
-_growbeg!(a::Vector, delta::Integer) =
+function _growbeg!(a::Vector, delta::Integer)
+    @assume_effects :terminates_globally
     ccall(:jl_array_grow_beg, Cvoid, (Any, UInt), a, delta)
-_growend!(a::Vector, delta::Integer) =
+end
+function _growend!(a::Vector, delta::Integer)
+    @assume_effects :terminates_globally
     ccall(:jl_array_grow_end, Cvoid, (Any, UInt), a, delta)
-_growat!(a::Vector, i::Integer, delta::Integer) =
+end
+function _growat!(a::Vector, i::Integer, delta::Integer)
+    @assume_effects :terminates_globally
     ccall(:jl_array_grow_at, Cvoid, (Any, Int, UInt), a, i - 1, delta)
+end
 
 # efficiently delete part of an array
 
-_deletebeg!(a::Vector, delta::Integer) =
+function _deletebeg!(a::Vector, delta::Integer)
+    @assume_effects :terminates_globally
     ccall(:jl_array_del_beg, Cvoid, (Any, UInt), a, delta)
-_deleteend!(a::Vector, delta::Integer) =
+end
+function _deleteend!(a::Vector, delta::Integer)
+    @assume_effects :terminates_globally
     ccall(:jl_array_del_end, Cvoid, (Any, UInt), a, delta)
-_deleteat!(a::Vector, i::Integer, delta::Integer) =
+end
+function _deleteat!(a::Vector, i::Integer, delta::Integer)
+    @assume_effects :terminates_globally
     ccall(:jl_array_del_at, Cvoid, (Any, Int, UInt), a, i - 1, delta)
+end
```

Comment:
We may first merge #48913 in and remove these annotations?

Reply:
Ah, sorry, I see you edited the comment.